src/view.c

```diff
@@ -866,6 +881,17 @@ view_update_app_id(struct view *view)
 		view->toplevel_handle, app_id);
 }
 
+void
+view_reload_ssd(struct view *view)
+{
+	assert(view);
+	if (view->ssd) {
```

Comment:
The existing logic (that `view_reload_ssd()` replaces) would skip over views that aren't mapped, and not recreate the SSD nodes for them. My understanding is that when we unmap a view, the entire scene tree (including the SSD nodes) is disabled, and so calling `ssd_destroy()/ssd_create()` is harmless, the nodes are invisible anyway. But I'd appreciate a second pair of eyes on this logic.

Reply:
The reason for skipping those and the reason for that 3rd allocated-but-disabled state was performance improvements.
That way we don't have to create everything from scratch (including uploading textures to the GPU by wlroots) when leaving fullscreen or toggling the SSD for another reason.

Due to the use of for example the autoscaling title buffer we may also have to wait for an enter event until pango renders the title in the new output scale.

Reply:
Re: reloading, if we skip unmapped views then we would need to set a flag to reload them next time they are mapped, right? Seems more straightforward to just do the reload immediately for all views that have SSD (mapped or not).

Re: toggling SSD, is it something that happens frequently enough to be worth caching those textures? I guess I'd vote for simplicity of implementation over performance here unless it's having a noticeable impact.

Reply:
> I guess I'd vote for simplicity of implementation over performance here unless it's having a noticeable impact.

I am fine with trying the destruct / recreate logic for now. If we detect that this will introduce lags on low power systems especially when un-fullscreening (e.g. video in a browser) we should be able to reintroduce the disable / enable flow again relatively easy.

We had issues before like for example the mouse cursor was starting to lag really bad when alt-tabbing out of a GPU intensive game which was caused by higher latency to upload the cursor texture to the GPU (and us not preventing to set the new cursor image to the one already set). At some points wlroots will likely add a feature to reuse textures for wlr_buffers that were bound to multiple wlr_scene_buffer nodes (e.g. all SSD icons and rounded corners) but as of right now each one of them gets a new texture uploaded for every SSD instance AFAIR.